### PR TITLE
feat(docs): Better docs for big chunk of schema stuff

### PR DESCRIPTION
**Pull request by freiksenet**
New docs for big chunk of schema stuff. sift docs need to be fully rewritten / changed to node-model docs. Inference stuff will be re-written by Vlad.

### CODE REVIEW

File: docs/docs/schema-generation.md [hunk 1]
```diff
@@ -2,66 +2,68 @@
 title: Schema Generation
 ---
 
-> This documentation isn't up to date with the latest [schema customization changes](/docs/schema-customization).
-> You can help by making a PR to [update this documentation](https://github.com/gatsbyjs/gatsby/issues/14228).
+Once the nodes have been sourced and transformed, the next step is to generate the GraphQL Schema. Gatsby Schema is different from many regular GraphQL schemas is that it's combines plugin or user defined schema information with data inferred from node shapes itself. The latter is called _schema inferrence_. Users or plugin can explicity define whole or part of the schema using [schema customization API](/docs/schema-customization). Usually every node will get a GraphQL Type based on its `node.internal.type` field. When using Schema Customization, all types that implement `Node` inteface become GraphQL Node Types and thus get root level fields for accessing them.
```
Comment (yg) [inline docs/docs/schema-generation.md:5]: Fix Grammar and Typo

```suggestion
Once the nodes have been sourced and transformed, the next step is to generate the GraphQL Schema. Gatsby Schema is different from many regular GraphQL schemas is that it combines plugin or user defined schema information with data inferred from node shapes itself. The latter is called _schema inferrence_. Users or plugin can explicitly define whole or part of the schema using [schema customization API](/docs/schema-customization). Usually every node will get a GraphQL Type based on its `node.internal.type` field. When using Schema Customization, all types that implement `Node` interface become GraphQL Node Types and thus get root level fields for accessing them.
```

File: docs/docs/schema-generation.md [hunk 2]
```diff
@@ -2,66 +2,68 @@
...
+### `setFieldsOnGraphQLType`
 
-This step is quite complex, but at its most basic, it infers GraphQL Fields by constructing an `exampleObject` that merges all fields of the type in Redux. It uses this to infer all possible fields and their types, and construct GraphQL versions of them. It does the same for fields created by plugins (like in step 1). This step is explained in detail in [GraphQL Node Types Creation](/docs/schema-gql-type).
+This allows adding GraphQL Fields to any node type. This operates on GraphQL types itself and syntax matches `graphql-js` field definitions. This API will be marked as deprecated in Gatsby v3, moved under flag in Gatsby v4 and removed from Gatsby v5. `createTypes` and `addResolvers` should solve all the use cases when this API is used.
 
-### 3. Create Input filters
+### `graphql-config.js` mapping
 
-This step creates GraphQL input filters for each field so the objects can be queried by them. More details in [Building the Input Filters](/docs/schema-input-gql).
+It's possible to connect types throught site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
```
Comment (yg) [inline docs/docs/schema-generation.md:37]: Typo

```suggestion
It's possible to connect types throughout site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
```

File: docs/docs/schema-generation.md [hunk 3]
```diff
@@ -2,66 +2,68 @@
...
+
+For each type, utility types are created. Those are input object types, used for searching, sorting and filtering, and types for paginated data (Connections and Edges).
+
+For searching and sorting, Gatsby goes through every field in the type and converts them to corresponding Input GraphQL types. Scalars are converted to objects with filter operator fields like "eq" or "ni", ObjectTypes are converted to InputObjectTypes. For sorting, enums are created out of all fields so that one can use that to specify the sort.
+
+Those types are used to create _root fields_ of the schema in `Query` type. For each node type a root field for querying one item and paginated items are created (eg for type BlogPost it would be `blogPost` and `allBlogPost`).
+
+## 6. Merging in third-party schemas
+
+If a plugin like `gatsby-source-graphql` is used, all third-party schemas that it provided are merged in into the Gatsby schema.
```
Comment (yg) [inline docs/docs/schema-generation.md:59]: Remove preposition

```suggestion
If a plugin like `gatsby-source-graphql` is used, all third-party schemas that it provided are merged into the Gatsby schema.
```

File: docs/docs/schema-root-fields.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    nodes {
+      id
+    }
+  }
+}
+```
+
+## Singular root fields
+
+Singular root fields accept same `filter` parameter as plural, but the `filter` is spread directily into arguments. Thus filter parameters need to be passed to the field directly. They return the resulting object directly.
```
Comment (yg) [inline docs/docs/schema-root-fields.md:29]: Typo

```suggestion
Singular root fields accept same `filter` parameter as plural, but the `filter` is spread directly into arguments. Thus filter parameters need to be passed to the field directly. They return the resulting object directly.
```

**Comment by freiksenet** [comment]
@AishaBlake Does removing the label means it's good to merge?

**Comment by AishaBlake** [comment]
Really sorry for the delay, @freiksenet! Removing the label just means it's been triaged.

File: docs/docs/schema-generation.md [hunk 5]
```diff
@@ -2,66 +2,68 @@
...
+This allows adding GraphQL Fields to any node type. This operates on GraphQL types itself and syntax matches `graphql-js` field definitions. This API will be marked as deprecated in Gatsby v3, moved under flag in Gatsby v4 and removed from Gatsby v5. `createTypes` and `addResolvers` should solve all the use cases when this API is used.
 
-### 3. Create Input filters
+### `graphql-config.js` mapping
 
-This step creates GraphQL input filters for each field so the objects can be queried by them. More details in [Building the Input Filters](/docs/schema-input-gql).
+It's possible to connect types throughout site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
 
-### 4. ProcessedTypeNode creation with resolve implementation
+It is a convenient escape hatch and there are no plans to deprecate this API at the moment.
```
[the remark this answers is not in the document]
Comment (freiksenet) [reply docs/docs/schema-generation.md]: Fixed the phrasing.

**Comment by freiksenet** [comment]
@AishaBlake Thanks a lot for the review! I've fixed/accepted issues that you've raised, added redirects and rebased. It's now ready for re-review and a merge.

**Comment by AishaBlake** [comment]
Thanks, @freiksenet! I see the linter is picking up on [a few things](https://circleci.com/gh/gatsbyjs/gatsby/476732?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). If you get to this before me, would you mind fixing those? If not, I'll take care of them. This is one of the first things on my "to review" list for next week.

**Comment by freiksenet** [comment]
@AishaBlake Fixed the linter issues now :)

**Pull request merged**